src/raftstore/store/fsm/apply.rs

```diff
@@ -3917,4 +3919,126 @@ mod tests {
         });
         res.unwrap_err();
     }
+
+    #[test]
+    fn test_apply_tilt() {
+        let (tx, _rx) = mpsc::channel();
+        let sender = Notifier::Sender(tx);
+        let (_tmp, engines) = create_tmp_engine("test-apply");
+        let host = Arc::new(CoprocessorHost::default());
+        let (_dir, importer) = create_tmp_importer("apply-apply");
+        let (region_scheduler, _snapshot_rx) = dummy_scheduler();
+        let mut cfg = Config::default();
+        cfg.messages_per_tick = 8;
+        let cfg = Arc::new(cfg);
+        let (router, mut system) = create_apply_batch_system(&cfg);
+        let builder = super::Builder {
+            tag: "test-store".to_owned(),
+            cfg,
+            coprocessor_host: host,
+            importer,
+            region_scheduler,
+            sender,
+            engines: engines.clone(),
+            router: router.clone(),
+        };
+        system.spawn("apply".to_owned(), builder);
+        let mut reg = Registration::default();
+        reg.id = 1;
+        reg.term = 1;
+        reg.region.set_id(1);
+        reg.region.set_end_key(b"k5".to_vec());
+        reg.apply_state.set_applied_index(3);
+        reg.region.mut_peers().push(new_peer(1, 2));
+        reg.region.mut_region_epoch().set_conf_ver(1);
+        reg.region.mut_region_epoch().set_version(3);
+
+        router.schedule_task(1, Msg::Registration(reg));
+
+        let mut reg = Registration::default();
+        reg.id = 2;
+        reg.term = 2;
+        reg.region.set_id(2);
+        reg.region.set_start_key(b"l5".to_vec());
+        reg.region.set_end_key(b"z5".to_vec());
+        reg.apply_state.set_applied_index(3);
+        reg.region.mut_region_epoch().set_version(3);
+        reg.region.mut_peers().push(new_peer(1, 2));
+        reg.region.mut_region_epoch().set_conf_ver(1);
+
+        router.schedule_task(2, Msg::Registration(reg));
+
+        let (send, rec) = channel();
+
+        let mut apply0_handle = 0;
+        let mut apply1_handle = 0;
+
+        let mut worker: Vec<JoinHandle<()>> = Vec::new();
+        for _i in 0..1 {
+            let tx1 = send.clone();
+            let router = router.clone();
+            let t = thread::spawn(move || {
+                for _i in 0..1000000 {
```

Comment:
This number of threads is scary, could we decrease it to a smaller number like 16?

Reply:
That is not a thread, we execute 1000000 cycles in a thread, but the sender of the channel will copy so many copies, but when the function execution ends, it will be automatically destroyed.